components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDAO.java

```diff
@@ -770,6 +827,115 @@ private void removeOauthOIDCPropertyTable(Connection connection,
+     * @param id         AppId of the OAuth app.
+     * @return List of scope validator names.
+     * @throws SQLException Sql error.
+     */
+    private String[] getScopeValidators(Connection connection, int id) throws SQLException {
+
+        List<String> scopeValidators = new ArrayList<>();
+        try (PreparedStatement stmt = connection.prepareStatement(SQLQueries.OAuthAppDAOSQLQueries
+                .GET_APP_SCOPE_VALIDATORS)) {
+            stmt.setInt(1, id);
+            try (ResultSet rs = stmt.executeQuery()) {
+                while (rs.next()) {
+                    scopeValidators.add(rs.getString(1));
+                }
+            }
+        }
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("Retrieving %d Scope validators registered for OAuth appId %d",
+                    scopeValidators.size(), id));
+        }
+        return scopeValidators.toArray(new String[scopeValidators.size()]);
+    }
+
+    /**
+     * Update the scope validator of OAuth app by remove all the registered scope validators and then add as new entry.
+     *
+     * @param connection      Same db connection used in OAuth update.
+     * @param appId           Id of consumerApp.
+     * @param scopeValidators List of scope validators.
+     * @throws SQLException Sql error.
+     */
+    private void updateScopeValidators(Connection connection, int appId, String[] scopeValidators)
+            throws SQLException {
+
+        log.debug(String.format("Removing  Scope validators registered for OAuth appId %d", appId));
+        try (PreparedStatement stmt = connection.prepareStatement(SQLQueries.OAuthAppDAOSQLQueries
+                .REMOVE_APP_SCOPE_VALIDATORS)) {
+            stmt.setInt(1, appId);
+            stmt.execute();
+        }
+        addScopeValidators(connection, appId, scopeValidators);
+    }
+
+    /**
+     * Get the application id using the client id.
+     *
+     * @param connection Same db connection used in OAuth creation.
+     * @param clientId   Client id of the created app.
+     * @return Application id of the client id.
+     * @throws SQLException                Sql exception.
+     * @throws InvalidOAuthClientException Invalid OAuth Client Exception.
+     * @throws IdentityOAuth2Exception     Identity OAuth2 Exception.
+     */
+    private int getAppIdByClientId(Connection connection, String clientId)
+            throws SQLException, InvalidOAuthClientException, IdentityOAuth2Exception {
+
+        int appId = 0;
+        try (PreparedStatement prepStmt = connection.prepareStatement(SQLQueries.OAuthAppDAOSQLQueries
+                .GET_APP_ID_BY_CONSUMER_KEY)) {
+            prepStmt.setString(1, persistenceProcessor.getProcessedClientId(clientId));
+            try (ResultSet rSet = prepStmt.executeQuery()) {
+                boolean rSetHasRows = false;
+                while (rSet.next()) {
+                    // There is at least one application associated with a given key
```

Comment:
Comments should end with a full stop. Please check in other places as well.

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/942/commits/ca4690d734709b108ef2df33a69ae3c9242febc8